crates/data_components/src/lib.rs

```diff
@@ -59,3 +64,25 @@ pub trait Stream {
         table_reference: OwnedTableReference,
     ) -> Result<Arc<dyn TableProvider + 'static>, Box<dyn Error + Send + Sync>>;
 }
+
+#[async_trait]
+pub trait DeleteTableProvider: TableProvider {
+    async fn delete(
```

Comment:
```suggestion
    async fn delete_from(
```

I think calling it `delete_from` is clearer